src/utils/lmsService.ts

```diff
@@ -107,3 +107,19 @@ export const embed = async (resource: Resource, mode: string, lmsPlatform: LMSTy
       break;
   }
 };
+
+export const getLMSSearchParams = () => {
+  const searchParams = new URLSearchParams(window.location.search);
+  const newSearchParams = new URLSearchParams();
+  if (searchParams.get(LMSParametersName.Navbar) === 'false') newSearchParams.append(LMSParametersName.Navbar, 'false');
+  if (searchParams.get(LMSParametersName.Footer) === 'false') newSearchParams.append(LMSParametersName.Footer, 'false');
+  if (searchParams.get(LMSParametersName.BBShowEmbedButton) === 'true')
+    newSearchParams.append(LMSParametersName.BBShowEmbedButton, 'true');
+  if (searchParams.get(LMSParametersName.EdxShowEmbedButton) === 'true')
+    newSearchParams.append(LMSParametersName.EdxShowEmbedButton, 'true');
+  if (searchParams.get(LMSParametersName.ItsLearningShowEmbedButton) === 'true')
+    newSearchParams.append(LMSParametersName.ItsLearningShowEmbedButton, 'true');
+  const ItsLearningReturnUrl = searchParams.get(LMSParametersName.ItsLearningReturnUrl);
+  if (ItsLearningReturnUrl) newSearchParams.append(LMSParametersName.ItsLearningReturnUrl, ItsLearningReturnUrl);
+  return newSearchParams;
```

Comment:
Finnes det en smartere måte å gjøre dette på?

Reply:
Funksjonen eksisterer for å kunne gå fra http://dlrnewqa2.s3-website-eu-west-1.amazonaws.com/?bbShowEmbedButton=true&inst=ntnu&inst=uib&query=fotosession

klikke på en av ressursene, og beholde bbShowEmbedButton=true i url'en:
http://dlrnewqa2.s3-website-eu-west-1.amazonaws.com/resource/2ee4177a-06ca-4883-8343-16670fad3e1a?bbShowEmbedButton=true

Reply:
Eksempel url'ene er til versjon med mock objekter, så bare å leke seg der :)

Reply:
Usikker på om jeg skjønner, men virker som du vil bygge opp en ny `URLSearchParams` som er helt lik den orginale?
I så fall kan du bare bruke den orginale med en gang, da jeg tipper du er nærmere mål på linje 112 enn på linje 113.
Ser ut som alt dette blir det samme
```
export const getLMSSearchParams = () => {
   return new URLSearchParams(window.location.search);
}
```

hvis du vil endre bare enkelte params, kan du bruke `set` eller `delete` i stedet for `append`

Reply:
Nei, jeg vasker bort en drøss queryParams som eksisterer fra før av, og det er like mange av de som det er de som jeg vil beholde. (eksempelvis ønsker jeg ikke å beholde inst og query)

Reply:
hm, da blir det nok litt knotete uansett ja. hvis du vil ha minst kode tenker jeg kanske noe sånn :
```
const keysToRemove = [inst, query, (...)]
const newSearchParams = new URLSearchParams(window.location.search);
keysToRemove.forEach(key => newSearchParams.delete(key));
```
evt vinkle det motsatt, med `keysToKeep` i stedet

Reply:
Supert tusen takk. Gikk for hvit-liste i stedet, tror jeg er nødt til å lage en ny URLSearchParams object, men nå er koden litt mer robust for endringer i parameter lista.